riscv-pages/src/state.rs

```diff
@@ -111,3 +111,18 @@ impl InternalData for InternalDirty {}
 impl Cleanable for InternalDirty {
     type Cleaned = InternalClean;
 }
+
+/// A shared (and implicitly dirty) page that has been assigned as a mapped page.
+#[derive(Debug)]
+pub enum SharedDirty {}
+
+/// A mapped page that is eligible for conversion to a Shared page.
+#[derive(Debug)]
+pub enum Shareable {}
+impl State for Shareable {}
+impl Converted for Shareable {}
+impl State for SharedDirty {}
+impl Mappable<MeasureOptional> for SharedDirty {}
```

Comment:
nit: I'd call this state `MappableShared` or (`MappableDirty`) rather than `SharedDirty`. Shared pages will always be dirty; the point we're trying to convey with this type name is that it's a shared page and it is now ready to be mapped.